web/src/games/cardtable/CardContainer.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FunctionComponent } from 'react';
+import { createStyles, makeStyles } from '@material-ui/core/styles';
+import GridList from '@material-ui/core/GridList';
+import GridListTile from '@material-ui/core/GridListTile';
+import { ICard } from './game';
+
+interface ICardContainerProps {
+  cards: ICard[];
+  deck?: boolean;
+  private?: boolean;
+  flipped?: boolean;
+  turn?: boolean;
+  name: string;
+  className?: string;
+}
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    root: {
+      width: '400px',
+      height: '80px',
+      fontSize: '10px',
+      display: 'flex',
+      flexWrap: 'nowrap',
+      justifyContent: 'center',
+      overflow: 'hide',
+      backgroundColor: 'grey',
+    },
+    gridList: {
+      flexWrap: 'nowrap',
+      height: '80px',
+      fontSize: '10px',
+      minWidth: '400px',
+      // Promote the list into his own layer on Chrome. This cost memory but helps keeping high FPS.
+      //transform: 'translateZ(0)',
+      alignContent: 'center',
+      backgroundColor: 'blue',
+    },
+
+    fitpicture: {
+      height: '80px',
+      objectFit: 'scale-down',
+      borderRadius: 10,
+    },
+  }),
+);
+
+/**
+ * The example data is structured as follows:
+ *
+ * import image from 'path/to/image.jpg';
+ * [etc...]
+ *
+ * const tileData = [
+ *   {
+ *     img: image,
+ *     title: 'Image',
+ *     author: 'author',
+ *   },
+ *   {
+ *     [etc...]
+ *   },
+ * ];
+ */
+const CardContainer: FunctionComponent<ICardContainerProps> = (props: ICardContainerProps) => {
+  const classes = useStyles();
```

Comment:
Ditto here, move to its own CSS file.

Reply:
> * Are the rules coded of Cribbage? Because this is a classic game, you could name your game "Cribbage", and it will help bring users from search engines. If you want to create other games in the future using the same UI and etc, you can re-use some components and still have the rules coded for each game.

The Card Table Game can be named **anything**, but it isn't a Cribbage game it's the first version of a card game simulator.  It's designed to support remote gameplay for the cumbersome abstract operations you execute when playing at home.  Shuffle deal, play tricks, quit tricks, are all operations that are exposed to the client.  To the extent that it won't impede players who wish to adapt their own style of play it will have many options to strip away the computer guardrails on rules.  Cribbage is a family favorite but we regularly meet and play three or four card games during a visit.  Think of cardtable as an attempt to facilitate such outings in an online community.

> * How are you planning for users to trigger actions ? I couldnt find buttons for triggering the actions available on the debug menu.  

There are more moves in the tip of feature/cardtable waiting to be merged back to you.  Currently it supports moveCard, cutDeck, deal, putToCrib, play, pegPoints, flipCrib. I played a full game in a test this evening and was pleased at how fluid the play was -- state diagrams mapping out the game are my next step -- there's a couple easy ones and many complicated but as I've mentioned these are early steps in a longer journey.  


> * Can you move the card component to gamesShared? I think this would be a useful component for other games too!
I will as the community agrees to the solution and finds value in it.  It's a labor of love, not my main line of development.  I mentioned one of your big successes is you remain approachable -- the barriers to integrating into such an effort are steep if you pick the wrong community.

> * Can you create some stories for your components? This is a form of documentation that will help other developers to understand and help with your game. As an example, you can check any file finishing with "storybook" on the name on other games folders.

I took a dive into Storybook and will try and keep my docs up-to-date.  I added a more representative thumbnail and a good deal about cribbage play on my description.md.  If you see something that needs improving on those first attempts give me some specific feedback.  I've scrolled through most of the Storybook collection, I can't see any details about any of the previous submissions.  Point out a good example and I'll try and emulate it -- All of the ones I encountered on the document set were pretty plain descriptions of components -- we also appear to be using only 1/2 of the framework -- there's a canvas tab on each of the doc nodes that is empty.  The doc framework looks interesting -- point me in that direction with some good docs, and I'll provide the lightest that we can agree to.


> * Are you using the blue background for debugging? It would be better if it was black in the final version. 

I am using color to help with my struggles in doing html/js/react.js ui solutions -- I have a hard time with CSS and it sometimes takes simplifications to see subtle problems with layout and display characteristics -- I learned the color tricks from a course on Web Dev by Colt Steele. I find it a good tool to debug UI.  Card tables are green usually, and that is one consideration, but a late one -- UI is decoration for the app.
  
> * The horizontal and vertical scrolling inside each container is probably not going to be something users are going to find easy to use. It would be better if we only allowed horizontal scrolling for each element, and a single vertical scroll for the whole page. Check this that might help:
>   https://www.freeboardgames.org/docs/?path=/story/infrastructure-common-carousel-carousel--mobile
>   https://www.freeboardgames.org/docs/?path=/story/infrastructure-common-carousel-carousel--desktop

Thanks, I'll look at those... I found the carousel docs in the Storybook offerings.  I couldn't tell if it was adaptive and responsive by design -- React.js has been in my sights for a while and while I'm still learning you'll see my experiments with various web design techniques.  Most of my React.js so far has been functional components, the same guy(s) who taught the web bootcamp (Colt Steel) is teaching me about a bunch more, contexts, hooks, redux, router. It's just a lot to get in one gulp -- but I like the component architecture a lot I was experimenting with components when a colleague asked me to investigate React.js -- I find the opinionated frameworks easier to learn, as long as you don't stray too much from their bread and butter, React uses semantic-ui natively (I think) and you guys are using another CSS framework.  That kind of impedance snowballs over time, it would be good to be declarative, but also provide the option of customization -- "batteries included, but optional" 

It's like drinking from Niagra Falls most days.  I planned on correcting the UI behavior after the state graph model was delivered.  Still a lot outstanding aside from the UI/UX.

Thanks for a great review -- recommend we move standard concerns to linter logic -- docs-- where new folks will appreciate the guidance and guardrails.  If I forget to get you the requested improvements please open issues so I can get the broader opinion of the community as I learn.